### PR TITLE
[Instrumentation.Process] Cleanup public APIs.

### PR DESCRIPTION
**Pull request by Yun-Ting**
## Changes
Cleanup public APIs.

* [X] Appropriate `CHANGELOG.md` updated for non-trivial changes

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.Process/MeterProviderBuilderExtensions.cs [hunk 1]
```diff
@@ -26,17 +26,17 @@ namespace OpenTelemetry.Metrics;
...
     /// <returns>The instance of <see cref="MeterProviderBuilder"/> to chain the calls.</returns>
     public static MeterProviderBuilder AddProcessInstrumentation(this MeterProviderBuilder builder) => AddProcessInstrumentation(builder, configure: null);
 
     /// <summary>
-    /// Enables runtime instrumentation.
+    /// Enables process instrumentation.
     /// </summary>
     /// <param name="builder"><see cref="MeterProviderBuilder"/> being configured.</param>
-    /// <param name="configure">Runtime metrics options.</param>
+    /// <param name="configure">Process metrics options.</param>
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Process/MeterProviderBuilderExtensions.cs:39]: this method can be removed for now as there is nothing to configure in ProcessInstrumentaionOptions

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.Process/MeterProviderBuilderExtensions.cs (hunk 1 above)
Comment (Yun-Ting) [reply src/OpenTelemetry.Instrumentation.Process/MeterProviderBuilderExtensions.cs]: removed

Comment (Kielek) [reply src/OpenTelemetry.Instrumentation.Process/MeterProviderBuilderExtensions.cs]: The option class seems to be also not needed.

**Comment by Kielek** [comment]
Please update changelog. It might be breaking change for part of the users.

File: src/OpenTelemetry.Instrumentation.Process/MeterProviderBuilderExtensions.cs (hunk 1 above)
Comment (Yun-Ting) [reply src/OpenTelemetry.Instrumentation.Process/MeterProviderBuilderExtensions.cs]: updated to use internal

File: src/OpenTelemetry.Instrumentation.Process/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Cleanup public API
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Process/CHANGELOG.md:5]: please explicitly state the API being removed, so its easy to follow changelog.

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.Process/CHANGELOG.md (hunk 2 above)
Comment (Yun-Ting) [reply src/OpenTelemetry.Instrumentation.Process/CHANGELOG.md]: Updated changelog.

**Pull request merged**